_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_vaapi.cpp

```diff
@@ -1429,6 +1443,37 @@ mfxStatus SetSkipFrame(
     return MFX_ERR_NONE;
 }
 
+mfxStatus VAAPIEncoder::FillPriorityBuffer(mfxPriority& priority)
+{
+    VAStatus vaSts;
+    memset(&m_priorityBuffer, 0, sizeof(VAContextParameterUpdateBuffer));
+    m_priorityBuffer.flags.bits.context_priority_update = 1;   //need to set by parameter
+
+    if(priority == MFX_PRIORITY_LOW)
+    {
+        m_priorityBuffer.context_priority.bits.priority = 0;
+    }
+    else if (priority == MFX_PRIORITY_HIGH)
+    {
+        m_priorityBuffer.context_priority.bits.priority = m_MaxContextPriority;
+    }
+    else
+    {
+        m_priorityBuffer.context_priority.bits.priority = m_MaxContextPriority/2;
+    }
+
+    vaSts = vaCreateBuffer(m_vaDisplay,
```

Comment:
This might fail if driver does not support setting the priority. You need to store driver caps somewhere and likely do nothing or return some warning if user tries to use this API.

Reply:
Done.